storehaus-postgresql/src/main/scala/com/twitter/storehaus/postgres/PostgresStore.scala

```diff
@@ -0,0 +1,178 @@
+  }
+
+  override def multiGet[K1 <: K](ks: Set[K1]): Map[K1, Future[Option[V]]] = {
+    val result = doGet(ks)(client)
+    ks.iterator.map { key =>
+      (key, result.map(_.get(key)))
+    }.toMap
+  }
+
+  override def put(kv: (K, Option[V])): Future[Unit] = {
+    implicit val cli = client
+    kv match {
+      case (key, Some(value)) => doUpsert(List((key, value))).unit
+      case (key, None) => doDelete(List(key)).unit
+    }
+  }
+
+  override def multiPut[K1 <: K](kvs: Map[K1, Option[V]]): Map[K1, Future[Unit]] = {
+    if (kvs.isEmpty) {
+      Map.empty
+    } else {
+      val result = doMultiput(kvs)(client)
+      kvs.mapValues(_ => result)
+    }
+  }
+
+  private def selectSQL[K1 <: K](ks: List[K1], forUpdate: Boolean): String = {
+    val sql = ks match {
+      case Nil => EMPTY_STRING
+      case x :: Nil => SELECT_SQL_PREFIX + s" = ${toEscString(x)}"
+      case res => SELECT_SQL_PREFIX + " IN " + res.map(toEscString).mkString("(", ", ", ")")
+    }
+    if(forUpdate) {
+      sql + " FOR UPDATE"
+    } else {
+      sql
+    }
+  }
+
+  /**
+   * Convert list of tuples (key, value) to String
+   * to insert in SQL query
+   */
+  private def valuesStmt(values: List[(K, V)]): String =
+    values.map {
+      case (key, value) => s"(${toEscString(key)}, ${toEscString(value)})"
+    }.mkString(", ")
+
+  private def upsertSQL(values: List[(K, V)]): String = {
+    // Insert keys and values in the table
+    // overwrite if key is already exists in table
+    // 'upsert' is available since PostgreSQL 9.5
+    // https://wiki.postgresql.org/wiki/UPSERT
+    if (values.nonEmpty) {
+      s"""$INSERT_SQL_PREFIX ${valuesStmt(values)}
+          |ON CONFLICT ($kCol) DO UPDATE SET $vCol = EXCLUDED.$vCol;
+          |""".stripMargin
+    } else {
+      EMPTY_STRING
+    }
+  }
+
+  private def insertSQL(values: List[(K, V)]): String = {
+    // simple insert, will fail if some keys exists
+    if (values.nonEmpty) {
+      s"$INSERT_SQL_PREFIX ${valuesStmt(values)}"
+    } else {
+      EMPTY_STRING
+    }
+  }
+
+  private def deleteSQL(keys: List[K]): String = {
+    keys match {
+      case Nil => EMPTY_STRING
+      case x :: Nil => DELETE_SQL_PREFIX + s" = ${toEscString(x)};"
+      case _ => DELETE_SQL_PREFIX + s" IN ${keys.map(toEscString).mkString("(", ", ", ")")};"
+    }
+  }
+
+  private def updateSQL(values: List[(K, V)]): String = values.map{
+      case (key, value) => s"UPDATE $table SET $vCol = ${toEscString(value)} WHERE $kCol = ${toEscString(key)};"
+    }.mkString("\n")
+
+  protected[postgres] def doGet[K1 <: K](k: K1)
+                               (implicit cli: Client): Future[Option[V]] =
+    doGet(Set(k)).map(_.values.headOption)
+
+  protected[postgres] def doGet[K1 <: K](ks: Set[K1],
+                                         forUpdate: Boolean = false)
+                                        (implicit cli: Client): Future[Map[K, V]] =
+    if (ks.isEmpty) {
+      Future(Map.empty)
```

Comment:
`Future.value(Map.empty)` is better since it avoids the call-by-name parameter, and since that is a constant, you might save it as a `private[this] val` and avoid the reallocation.